pkg/operator/encryption/helpers_test.go

```diff
@@ -0,0 +1,271 @@
+			Conditions: []corev1.PodCondition{
+				{
+					Type:   corev1.PodReady,
+					Status: corev1.ConditionTrue,
+				},
+			},
+		},
+	}
+}
+
+func createDummyKubeAPIPodInUnknownPhase(name, namespace string) *corev1.Pod {
+	p := createDummyKubeAPIPod(name, namespace)
+	p.Status.Phase = corev1.PodUnknown
+	return p
+}
+
+func secretDataToEncryptionConfig(secret *corev1.Secret) (*apiserverconfigv1.EncryptionConfiguration, error) {
+	rawEncryptionConfig, exist := secret.Data[encryptionConfSecretForTest]
+	if !exist {
+		return nil, errors.New("the secret doesn't contain an encryption configuration")
+	}
+
+	decoder := apiserverCodecs.UniversalDecoder(apiserverconfigv1.SchemeGroupVersion)
+	decodedEncryptionConfig, err := runtime.Decode(decoder, rawEncryptionConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	encryptionConfig, ok := decodedEncryptionConfig.(*apiserverconfigv1.EncryptionConfiguration)
+	if !ok {
+		return nil, fmt.Errorf("encryption config has wrong type %T", decodedEncryptionConfig)
+	}
+	return encryptionConfig, nil
+}
+
+func validateActionsVerbs(actualActions []clientgotesting.Action, expectedActions []string) error {
+	if len(actualActions) != len(expectedActions) {
+		return fmt.Errorf("expected to get %d actions but got %d, got=%v, expected=%v", len(expectedActions), len(actualActions), actionStrings(actualActions), expectedActions)
+	}
+	for i, a := range actualActions {
+		if got, expected := actionString(a), expectedActions[i]; got != expected {
+			return fmt.Errorf("at %d got %s, expected %s", i, got, expected)
+		}
+	}
+	return nil
+}
+
+func actionString(a clientgotesting.Action) string {
+	return a.GetVerb() + ":" + a.GetResource().Resource + ":" + a.GetNamespace()
+}
+
+func actionStrings(actions []clientgotesting.Action) []string {
+	res := make([]string, 0, len(actions))
+	for _, a := range actions {
+		res = append(res, actionString(a))
+	}
+	return res
+}
+
+func createEncryptionCfgNoWriteKey(keyID string, keyBase64 string, resources ...string) *apiserverconfigv1.EncryptionConfiguration {
+	return &apiserverconfigv1.EncryptionConfiguration{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "EncryptionConfiguration",
+			APIVersion: "apiserver.config.k8s.io/v1",
+		},
+		Resources: []apiserverconfigv1.ResourceConfiguration{
+			{
+				Resources: resources,
+				Providers: []apiserverconfigv1.ProviderConfiguration{
+					{
+						Identity: &apiserverconfigv1.IdentityConfiguration{},
+					},
+					{
+						AESCBC: &apiserverconfigv1.AESConfiguration{
+							Keys: []apiserverconfigv1.Key{
+								{Name: keyID, Secret: keyBase64},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func createEncryptionCfgWithWriteKey(keysResources []encryptionKeysResourceTuple) *apiserverconfigv1.EncryptionConfiguration {
+	configurations := []apiserverconfigv1.ResourceConfiguration{}
+	for _, keysResource := range keysResources {
+		// TODO allow secretbox -> not sure if encryptionKeysResourceTuple makes sense
```

Comment:
nit: we could remove it since `secretbox` is disabled anyway - https://github.com/enj/cluster-kube-apiserver-operator/blob/enj/f/enc_config_ref7/pkg/operator/encryption/key_controller_encryption.go#L248